### PR TITLE
Rename go app

### PR DESCRIPTION
**Pull request by hervenicol**
Towards https://github.com/giantswarm/roadmap/issues/3226

This renames the project, I will check for obsolete loki references in a next PR.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ workflows:
      - architect/push-to-registries:
         context: architect
         name: push-to-registries
-        image: "giantswarm/loki-multi-tenant-proxy-gs"
```
Comment (QuentinBisson) [inline .circleci/config.yml:13]: I think we can do without the -gs suffix?

File: build/package/Dockerfile-with-tests [hunk 2]
```diff
@@ -7,9 +7,9 @@ ENV GO111MODULE=on
 ENV CGO_ENABLED=0
 
 RUN apk add  -U --no-cache git ca-certificates && \
-    mkdir -p src/github.com/k8spin/loki-multi-tenant-proxy
+    mkdir -p src/github.com/k8spin/grafana-multi-tenant-proxy
```
Comment (QuentinBisson) [inline build/package/Dockerfile-with-tests:10]:
```suggestion
    mkdir -p src/github.com/giantswarm/grafana-multi-tenant-proxy
```

File: .circleci/config.yml (hunk 1 above)
Comment (hervenicol) [reply .circleci/config.yml:13]: done :heavy_check_mark:

File: build/package/Dockerfile-with-tests (hunk 2 above)
Comment (hervenicol) [reply build/package/Dockerfile-with-tests]: Fixed, here and in various other places.

**Review by QuentinBisson — APPROVED** [review]

**Comment by hervenicol** [comment]
> Do we still need to keep the authors as we did a lot of changes?

Maybe not, but that's probably not in the scope of this PR.

**Pull request merged**